src/ui/components/DevTools.tsx

```diff
@@ -216,6 +219,24 @@ function _DevTools({
     }
   }, [loadingFinished, trackLoadingIdleTime, sessionId]);
 
+  useEffect(() => {
+    if (!userLoading && recording) {
+      setDefaultTags({
+        recording: {
+          id: recording.id,
+          title: recording.title,
+          url: recording.url,
+          userId: recording.user?.id,
+          workspace: recording?.workspace,
+          metadata: recording.metadata && {
+            testEnvironment: recording.metadata?.test?.environment,
```

Comment:
Not sure why `test.environment` is not typed...

Here's the runtime value
```
{
    "errors": [],
    "testRunner": {
        "name": "cypress",
        "version": "12.8.1"
    },
    "pluginVersion": "1.0.1"
}
```

Reply:
It is typed, but there are multiple versions of the data that require refinement..

https://github.com/replayio/devtools/blob/01bda21af5a12dd5808ca0ca6b1d3b5f37876401/packages/shared/graphql/types.ts#L249C14-L253

https://github.com/replayio/devtools/blob/01bda21af5a12dd5808ca0ca6b1d3b5f37876401/packages/shared/test-suites/RecordingTestMetadata.ts#L288-L296

Reply:
Not sure how to work with this... My TSC are not the best

Reply:
Metadata is stored differently depending on which version of the plug-in/backend stored it. Those types describe the various formats. If you try to access a property of one of the "any" types you might end up with `any`.

There are a bunch of helper functions further down that file for working with the different formats in a type-safe way:

https://github.com/replayio/devtools/blob/01bda21af5a12dd5808ca0ca6b1d3b5f37876401/packages/shared/test-suites/RecordingTestMetadata.ts#L833-L951C2